bundle/src/main/java/com/adobe/acs/commons/packaging/impl/AssetPackagerServletImpl.java

```diff
@@ -0,0 +1,281 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+package com.adobe.acs.commons.packaging.impl;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+import javax.jcr.RepositoryException;
+
+import com.adobe.acs.commons.packaging.PackageHelper;
+import com.day.cq.dam.api.DamConstants;
+import com.day.cq.dam.commons.util.DamUtil;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.felix.scr.annotations.Reference;
+import org.apache.felix.scr.annotations.sling.SlingServlet;
+import org.apache.jackrabbit.vault.fs.api.PathFilterSet;
+import org.apache.jackrabbit.vault.fs.filter.DefaultPathFilter;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.api.wrappers.ValueMapDecorator;
+
+/**
+ * ACS AEM Commons - Asset Packager Servlet
+ * Servlet end-point used to create packages of pages and the referenced assets.
+ */
+@SuppressWarnings("serial")
+@SlingServlet(
+        methods = {"POST"},
+        resourceTypes = {"acs-commons/components/utilities/packager/asset-packager"},
+        selectors = {"package"},
+        extensions = {"json"}
+)
+public class AssetPackagerServletImpl extends AbstractPackagerServlet {
+
+    /* Property names */
+    private static final String PN_PAGE_PATH = "pagePath";
+    private static final String PN_EXCLUDE_PAGES = "excludePages";
+    private static final String PN_ASSET_PREFIX = "assetPrefix";
+    private static final String PN_PAGE_EXCLUSIONS = "pageExclusions";
+    private static final String PN_ASSET_EXCLUSIONS = "assetExclusions";
+
+    /* Default Properties */
+    private static final String DEFAULT_PACKAGE_NAME = "assets";
+    private static final String DEFAULT_PACKAGE_GROUP_NAME = "Assets";
+    private static final String DEFAULT_PACKAGE_DESCRIPTION = "Asset Package initially defined by a ACS AEM Commons - "
+            + "Asset Packager configuration.";
+    private static final String THUMBNAIL_RESOURCE_PATH =
+            "/apps/acs-commons/components/utilities/packager/asset-packager/definition/package-thumbnail.png";
+
+    @Reference
+    private PackageHelper packageHelper;
+
+    private String customPrefix;
```

Comment:
Servlets are not threadsafe, and should therefore not use instance variables.  Though it's a low risk on a servlet like this that have very little chance of being executed simultaneously by two different users, out of principle we should update this to avoid the bad practice.

Reply:
I did this to avoid passing each of these 4 objects around in every method call but I see the point in changing that. Not sure I have a ton of options other than just passing them around either - it feels like I could do that better. Any suggestions?

Reply:
Probably best suggestion would be to modularize to a separate class that is instantiated on each request and executes the logic.